src/transpilation/resolve.ts

```diff
@@ -93,6 +98,60 @@ class ResolutionContext {
         }
     }
 
+    private resolveDependencyPathsWithPlugins(required: ProcessedFile, dependency: string) {
+        const requiredFromLuaFile = required.fileName.endsWith(".lua");
+        const dependencyPath = requiredFromLuaFile ? luaRequireToPath(dependency) : dependency;
+
+        for (const plugin of this.plugins) {
+            if (plugin.moduleResolution != null) {
+                const pluginResolvedPath = plugin.moduleResolution(
+                    dependency,
+                    dependencyPath,
+                    this.options,
+                    this.emitHost
+                );
+                if (pluginResolvedPath !== undefined) {
+                    // If lua file is in node_module
+                    if (requiredFromLuaFile && isNodeModulesFile(required.fileName)) {
+                        // If requiring file is in lua module, try to resolve sibling in that file first
+                        const resolvedNodeModulesFile = this.resolveLuaDependencyPathFromNodeModules(
+                            required,
+                            pluginResolvedPath
+                        );
+                        if (resolvedNodeModulesFile) {
+                            if (this.options.tstlVerbose) {
+                                console.log(
+                                    `Resolved file path for module ${dependency} to path ${dependencyPath} using plugin.`
+                                );
+                            }
+                            return resolvedNodeModulesFile;
+                        }
+                    }
+
+                    const isRelative = ["/", "./", "../"].some(p => pluginResolvedPath.startsWith(p));
+
+                    // // If the import is relative, always resolve it relative to the requiring file
+                    // // If the import is not relative, resolve it relative to options.baseUrl if it is set
+                    const fileDirectory = path.dirname(required.fileName);
+                    const relativeTo = isRelative ? fileDirectory : this.options.baseUrl ?? fileDirectory;
+
+                    // // Check if file is a file in the project
+                    const resolvedPath = path.join(relativeTo, pluginResolvedPath);
+                    const fileFromPath = this.getFileFromPath(resolvedPath);
```

Comment:
This looks just straight up duplicate from logic in `resolveDependencyPath`, please move to a shared function they both can use